src/synonyms.ts

```diff
@@ -15,40 +17,70 @@ export default async function synonyms({
     method: 'GET',
     uri: url_symbol as string,
     json: true
-  };
+    };
 
   const result: any = await _request(options_symbol)
     .then((res: any) => {
       if (res.response.docs.length === 1 && typeof res.response.docs[0] !== 'undefined') {
           return res.response.docs[0].alias_symbol.concat(res.response.docs[0].symbol);
-      }else {
-        const json_alias = retrieveAliasSynonyms({ symbol_alias : symbol });
-        return json_alias;
+      } else {
+        return aliasSynonyms({ symbol_alias : symbol });
       }
     })
     .catch((err: any) => {
       //console.log('_request error: ' + err);
       throw new Error('_request error: ' + err);
     });
 
-  console.log(result);
   return result;
 }
 
-function retrieveAliasSynonyms({ symbol_alias }: { symbol_alias: string }) {
+export async function allSynonyms() {
+  const url_all = 'http://rest.genenames.org/fetch/status/Approved';
+
+  const options_all = {
+    method: 'GET',
+    uri: url_all as string,
+    json: true
+    };
+
+  const json_all: any = _request(options_all)
+    .then((res: any) => {
+      const filterUndefined = R.pipe(R.prop('alias_symbol'), R.isNil, R.not);
+      const definedAlias = R.filter(filterUndefined, res.response.docs);
+      const reduceToSymbol = R.pick(['symbol','alias_symbol']);
+      const reducedAlias = R.map(reduceToSymbol, definedAlias);
+      const combinedReducedAlias = R.map(R.values, reducedAlias);
+      return R.map(R.flatten, combinedReducedAlias);
```

Comment:
You could probably use R.pipe here, but it will be a lot of Ramda! So perhaps it is actually clearer this way.

Reply:
Done